src/sparseml/optim/helpers.py

```diff
@@ -569,6 +574,44 @@ def _check_warn_dict_difference(original_dict, new_dict):
     return new_dict
 
 
+def add_framework_metadata(metadata: Dict[str, Dict]) -> Dict[str, Dict]:
```

Comment:
```suggestion
def add_framework_metadata(metadata: Dict[str, Dict], **extra_metadata) -> Dict[str, Dict]:
```
let's do something like this where each framework will pass in their specific version and then update the metadata dict with this metadata as well

i.e. for pytorch it will pass `add_framework_metadata(metadata, torch=torch.__version__)`